app/src/main/java/uy/com/bix/app/smsproject/activity/SettingsActivity.java

```diff
@@ -23,17 +205,146 @@ public boolean onCreateOptionsMenu(Menu menu) {
 	}
 
 	@Override
-	public boolean onOptionsItemSelected(MenuItem item) {
-		// Handle action bar item clicks here. The action bar will
-		// automatically handle clicks on the Home/Up button, so long
-		// as you specify a parent activity in AndroidManifest.xml.
-		int id = item.getItemId();
+	public void onSharedPreferenceChanged(SharedPreferences sharedPreferences,
+																				String key) {
+		updatePrefSummary(findPreference(key));
+	}
+
+	private void initSummary(Preference p) {
+		if (p instanceof PreferenceGroup) {
+			PreferenceGroup pGrp = (PreferenceGroup) p;
+			for (int i = 0; i < pGrp.getPreferenceCount(); i++) {
+				initSummary(pGrp.getPreference(i));
+			}
+		}
+		else {
+			updatePrefSummary(p);
+		}
+	}
+
+	private void updatePrefSummary(Preference p) {
+		String key = p.getKey();
+		if (p instanceof EditTextPreference) {
+			EditTextPreference editTextPref = (EditTextPreference) p;
+			if (editTextPref.getText() == null || editTextPref.getText().equals("")) {
+				switch(p.getKey()) {
+					case MESSAGE_KEY:
+						p.setSummary("Mensaje a enviar");
+						break;
+					case MAX_KEY:
+						p.setSummary("Tope maximo de mensajes a enviar");
+						break;
+					case NUMBER_KEY:
+						p.setSummary("Numero destinatario de la donacion");
+						break;
+				}
+			}
+			else {
+				p.setSummary(editTextPref.getText());
+			}
+		}
+		else if (key.equals(DATE_KEY)) {
+			p.setSummary(String.valueOf(expirationDay) + "/" +
+				String.valueOf(expirationMonth + 1) + "/" +
+				String.valueOf(expirationYear));
+		}
+		else if (key.equals(TIME_KEY)) {
+			p.setSummary(String.valueOf(expirationHour) + ":" + String.valueOf(expirationMinute));
+		}
+
+	}
+
+	public void saveSettings() {
+
+		// We save the user preferences using the default shared preferences
+		SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(contextOfApplication);
+		SharedPreferences.Editor editor = settings.edit();
+		editor.putInt("Year", expirationYear);
+		editor.putInt("Month", expirationMonth + 1);
+		editor.putInt("Day", expirationDay);
+		editor.putInt("Hour", expirationHour);
+		editor.putInt("Minute", expirationMinute);
+		editor.apply();
+
+		DateTime sendingDate = DateTime.now();
+		sendingDate = sendingDate.withMinuteOfHour(expirationMinute);
+		sendingDate = sendingDate.withHourOfDay(expirationHour);
+		sendingDate = sendingDate.withYear(expirationYear);
+		sendingDate = sendingDate.withMonthOfYear(expirationMonth + 1);
+		sendingDate = sendingDate.withDayOfMonth(expirationDay);
+
+		// The date must be in milliseconds
+		long whenToFireTask = sendingDate.getMillis();
+
+		isActive = settings.getBoolean("Active", false);
+		if (isActive) {
+			scheduleAlarm(whenToFireTask);
+			System.out.println(sendingDate);
```

Comment:
The system out are left in purpose to debug and test the app, they'll be deleted in the future